clickhouse-admin/src/context.rs

```diff
@@ -3,26 +3,50 @@
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
 use crate::{ClickhouseCli, Clickward};
+
+use anyhow::{anyhow, bail, Result};
+use camino::Utf8PathBuf;
+use clickhouse_admin_types::{
+    CLICKHOUSE_KEEPER_CONFIG_DIR, CLICKHOUSE_KEEPER_CONFIG_FILE,
+    CLICKHOUSE_SERVER_CONFIG_DIR, CLICKHOUSE_SERVER_CONFIG_FILE,
+};
 use omicron_common::address::CLICKHOUSE_TCP_PORT;
+use omicron_common::api::external::Generation;
 use oximeter_db::Client as OximeterClient;
 use slog::Logger;
+use std::fs::File;
+use std::io::{BufRead, BufReader};
 use std::net::SocketAddrV6;
+use std::str::FromStr;
 use std::sync::Arc;
 use tokio::sync::Mutex;
 
 pub struct KeeperServerContext {
     clickward: Clickward,
     clickhouse_cli: ClickhouseCli,
     log: Logger,
+    pub generation: Mutex<Option<Generation>>,
 }
 
 impl KeeperServerContext {
-    pub fn new(clickhouse_cli: ClickhouseCli) -> Self {
-        let log = clickhouse_cli
-            .log
-            .new(slog::o!("component" => "KeeperServerContext"));
+    pub fn new(
+        log: &Logger,
+        binary_path: Utf8PathBuf,
+        listen_address: SocketAddrV6,
+    ) -> Result<Self> {
+        let clickhouse_cli =
+            ClickhouseCli::new(binary_path, listen_address, log);
```

Comment:
Refactor as well, there was no need to pass clickhouse_cli as a parameter, but not clickward etc.